cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -160,9 +160,6 @@ cdef class Buffer(_cyBuffer, MemoryResourceAttributes):
                 s = <cyStream>stream
             self._mr._deallocate(self._ptr, self._size, s)
             self._ptr = 0
```

Comment:
This was something I had mentioned elsewhere, that we shouldn't be accessing object properties in a `__dealloc__` code path.

Why is `_ptr` still being touched here?

From the Cython docs on `__dealloc__`:

> In particular, don’t call any other methods of the object or do anything which might cause the object to be resurrected. It’s best if you stick to just deallocating C data.

Why not just do the minimum C-API invocation required in `__dealloc__` and duplicate whatever you need to from `close`?

Then we're not going against the documented behavior and causing a crashing knowing that we could have avoided it all along.